File: chazm-model/src/main/java/runtimemodels/chazm/model/event/PolicyEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.Policy;

/**
 * The {@linkplain PolicyEventFactory} interface defines the API for constructing {@linkplain PolicyEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PolicyEventFactory {

    /**
     * Constructs an {@linkplain PolicyEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param policy   the {@linkplain Policy}.
     * @return a {@linkplain PolicyEvent}.
     */
    PolicyEvent build(EventType category, Policy policy);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/PolicyEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.Policy

/**
 * The [PolicyEventFactory] interface defines the API for constructing [PolicyEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PolicyEventFactory {

    /**
     * Constructs an [PolicyEvent].
     *
     * @param category the [EventType].
     * @param policy   the [Policy].
     * @return a [PolicyEvent].
     */
    fun build(category: EventType, policy: Policy): PolicyEvent

}
